azure-devops/azext_devops/dev/repos/policy.py

```diff
@@ -0,0 +1,371 @@
+        return policy_client.delete_policy_configuration(project=project, configuration_id=id)
+    except VstsServiceError as ex:
+        raise CLIError(ex)
+
+# pylint: disable=too-many-locals
+def create_policy(repository_id, branch,
+                  isBlocking=False, isEnabled=False,
+                  policy_type=None,
+                  minimumApproverCount=None, creatorVoteCounts=None, allowDownvotes=None, resetOnSourcePush=None,
+                  useSquashMerge=None,
+                  buildDefinitionId=None, queueOnSourceUpdateOnly=None, manualQueueOnly=None, displayName=None, validDuration=None,
+                  maximumGitBlobSizeInBytes=None, useUncompressedSize=None,
+                  optionalReviewerIds=None, requiredReviewerIds=None, message=None,
+                  organization=None, project=None, detect=None):
+    """
+    :param repository_id: Id (UUID) of the repository on which to apply the policy
+    :type repository_id: string
+    :param branch: Branch on which this policy should be applied
+    :type branch: string
+    :param isBlocking: Tells if the policy should be blocking or not
+    :type isBlocking: bool
+    :param isEnabled: Tells if the policy is enabled or not
+    :type isEnabled: bool
+    :param policy_type: Type of policy you want to create
+    :type policy_type: string
+
+    :param optionalReviewerIds: Optional Reviewers (List of email ids seperated with ';'). Required if policy type is RequiredReviewersPolicy.
+    :type optionalReviewerIds: string
+    :param requiredReviewerIds: Required Reviewers (List of email ids seperated with ';'). Required if policy type is RequiredReviewersPolicy.
+    :type requiredReviewerIds: string
+    :param message: Message. Required if policy type is RequiredReviewersPolicy.
+    :type message: string
+
+    :param minimumApproverCount: Minimum approver count. Required if policy type is ApproverCountPolicy.
+    :type minimumApproverCount: int
+    :param creatorVoteCounts: Creator vote counts or not. Required if policy type is ApproverCountPolicy
+    :type creatorVoteCounts: bool
+    :param allowDownvotes: Allow Downvotes. Required if policy type is ApproverCountPolicy.
+    :type allowDownvotes: bool
+    :param resetOnSourcePush: Reset on Source Push. Required if policy type is ApproverCountPolicy.
+    :type resetOnSourcePush: bool
+
+    :param buildDefinitionId: Build Definition Id. Required if policy type is Buildpolicy.
+    :type buildDefinitionId: int
+    :param queueOnSourceUpdateOnly: Queue Only on source update. Required if policy type is Buildpolicy.
+    :type queueOnSourceUpdateOnly: bool
+    :param manualQueueOnly : Manual Queue Only. Required if policy type is Buildpolicy.
+    :type manualQueueOnly : bool
+    :param displayName : Display Name. Required if policy type is Buildpolicy.
+    :type displayName : string
+    :param validDuration : Valid duration. Required if policy type is Buildpolicy.
```

Comment:
Policy validity duration. Accepted format in days. For example, --validDuration 3